app/controllers/dish_ingredients_controller.rb

```diff
@@ -0,0 +1,12 @@
+class DishIngredientsController < ApplicationController
+
+  def new
+    DishIngredient.create(dish_id: params[:id], ingredient_id: params[:add_ingredient])
+    redirect_to "/dishes/#{params[:id]}"
+  end
```

Comment:
This is great, only one thing. This should be a `create` action. Remember that the `new` action is for a `GET` request to retrieve a form for the view. The code is perfect, though.